src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -151,12 +151,19 @@ private ushort Significand
 
         public static bool operator ==(Half left, Half right)
         {
-            return left.Equals(right);
+            if (IsNaN(left) || IsNaN(right))
+            {
+                // IEEE defines that NaN is not equal to anything, including itself.
+                return false;
+            }
+
+            // IEEE defines that positive and negative zero are equivalent.
+            return (left._value == right._value) || AreZero(left, right);
         }
 
         public static bool operator !=(Half left, Half right)
         {
-            return !(left.Equals(right));
+            return !(left == right);
```

Comment:
Are you missing a test for the change on this line?

Reply:
I'll add a test for this and push it to this branch